naptime-graphql/src/test/scala/org/coursera/naptime/ari/graphql/schema/NaptimeUnionFieldTest.scala

```diff
@@ -42,17 +42,22 @@ class NaptimeUnionFieldTest extends AssertionsForJUnit with MockitoSugar {
   when(schemaMetadata.getResourceOpt(resourceName)).thenReturn(Some(resource))
   when(schemaMetadata.getSchema(resource)).thenReturn(Some(null))
 
-  def buildUnionDataSchema(types: List[DataSchema], typedDefinitions: Option[Map[String, String]] = None): UnionDataSchema = {
+  def buildUnionDataSchema(
+      types: List[DataSchema],
+      typedDefinitions: Map[String, String] = Map.empty,
+      properties: Map[String, AnyRef] = Map.empty): UnionDataSchema = {
     val union = new UnionDataSchema()
     val stringBuilder = new java.lang.StringBuilder()
     union.setTypes(types.asJava, stringBuilder)
-    typedDefinitions.foreach(td => union.setProperties(
-      Map("typedDefinition" -> td.asJava.asInstanceOf[AnyRef]).asJava))
+    val modifiableMap = collection.mutable.Map[String, AnyRef]()
+    modifiableMap.getOrElseUpdate("typedDefinition", typedDefinitions.asJava.asInstanceOf[AnyRef])
```

Comment:
why is this `getOrElseUpdate` -- it looks like this is always going to be on an empty map. 

could you instead do:
```scala
val unionProperties = Map("typedDefinition" -> typedDefinitions.asJava.asInstanceOf[AnyRef]) ++ properties
union.setProperties(unionProperties)